SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/BruteForceDetection.h

```diff
@@ -61,18 +80,117 @@ class SOFA_SOFABASECOLLISION_API BruteForceDetection :
     void init() override;
     void reinit() override;
 
-    void addCollisionModel (core::CollisionModel *cm) override;
-    void addCollisionPair (const std::pair<core::CollisionModel*, core::CollisionModel*>& cmPair) override;
+    //////////////////////////////
+    /// BROAD PHASE interface ///
+    //////////////////////////////
+
+    /** \brief In the broad phase, ignores collision with the provided collision model if possible and add pairs of
+     * collision models if in intersection.
+     *
+     * Ignore the collision with the provided collision model if it does not intersect with the box defined in
+     * the Data box when it is defined.
+     * Add the provided collision model to be investigated in the narrow phase in case of self collision.
+     * Check intersection with already added collision models. If it can intersect another collision model, the pair
+     * is added to be further investigated in the narrow phase.
+     */
+    void addCollisionModel(core::CollisionModel *cm) override;
+
+    void beginBroadPhase() override;
+    void endBroadPhase() override;
+
+    //////////////////////////////
+    /// NARROW PHASE interface ///
+    //////////////////////////////
+
+    /** \brief In the narrow phase, examine a potential collision between a pair of collision models, which has
+     * been detected in the broad phase.
+     *
+     * The function traverses the hierarchy of CollisionElement's contained in each CollisionModel to avoid
+     * unnecessary pair intersections.
+     * An iterative form of the hierarchy traversal is adopted instead of a recursive form.
+     */
+    void addCollisionPair(const std::pair<core::CollisionModel*, core::CollisionModel*>& cmPair) override;
 
-    void beginBroadPhase() override
-    {
-        core::collision::BroadPhaseDetection::beginBroadPhase();
-        collisionModels.clear();
-    }
 
     void draw(const core::visual::VisualParams* /* vparams */) override { }
 
-    inline bool needsDeepBoundingTree()const override {return true;}
+    inline bool needsDeepBoundingTree() const override { return true; }
+
+protected:
+
+    /// Return true if the provided CollisionModel intersect boxModel, false otherwise
+    bool intersectWithBoxModel(core::CollisionModel *cm) const;
+
+    /// Return true if the provided CollisionModel can collide with itself
+    bool doesSelfCollide(core::CollisionModel *cm) const;
+
+    /// Return true if both collision models belong to the same object, false otherwise
+    static bool isSelfCollision(core::CollisionModel* cm1, core::CollisionModel* cm2);
+
+    /// Build a list of TestPair's from internal and external children of two CollisionModel's
+    static void initializeExternalCells(
+            core::CollisionModel *cm1,
+            core::CollisionModel *cm2,
+            std::queue<TestPair>& externalCells);
+
+    /// Store data related to two finnest CollisionModel's
+    struct FinnestCollision
```

Comment:
Finest with one 'n' ...

Reply:
Good catch! And it's not the only place with this typo